token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -330,9 +339,25 @@ fn process_deposit(
             return Err(TokenError::ConfidentialTransferDepositsAndTransfersDisabled.into());
         }
 
-        destination_confidential_transfer_account.pending_balance = ops::add_to(
-            &destination_confidential_transfer_account.pending_balance,
-            amount,
+        if u64::from(destination_confidential_transfer_account.pending_balance_credit_counter)
+            >= u64::from(
```

Comment:
```suggestion
            > u64::from(
```
nit, but to me `maximum_pending_balance_credit_counter` implies the max that is acceptable